electrum/plugins/bitbox02/bitbox02.py

```diff
@@ -0,0 +1,401 @@
+        self.handler = handler
+
+        self._expected_device = None
+        bitboxes = devices.get_any_bitbox02()
+        if not bitboxes:
+            raise "No BitBox02 detected"
+
+    def label(self):
+        return "BitBox02"
+
+    def is_initialized(self):
+        return True
+
+    def close(self):
+        try:
+            self.device.close()
+        except:
+            pass
+
+    def has_usable_connection_with_device(self):
+        bitboxes = devices.get_any_bitbox02()
+        if not bitboxes:
+            return False
+        return True
+
+    def pairing_dialog(self, bitboxes):
+
+        def show_pairing(code: str) -> None:
+            msg = "Please compare and confirm the pairing code on your BitBox02:" + "\n"
+            choice = [code]
+            try:
+                reply = self.handler.win.query_choice(msg, choice)
+            except Exception:
+                return False# Back button pushed
+            return True
+
+        def attestation_check(result: bool) -> None:
+            if result:
+                print("Device attestation PASSED")
+            else:
+                print("Device attestation FAILED")
+
+        hid_device = hid.device()
+        hid_device.open_path(bitboxes["path"])
+
+        self.device = bitbox02.BitBox02(
+            transport = u2fhid.U2FHid(hid_device),
+            device_info=bitboxes,
+            show_pairing_callback=show_pairing,
+            attestation_check_callback=attestation_check,
+        )
+
+    def bip32_list_from_keypath_string(self, bip32_path):
+        derivations = bip32_path.split('/')
+        is_mainnet = True
+        print(bip32_path)
+        if len(derivations) > 2:
+            if derivations[2] == '1\'':
+                is_mainnet = False
+        keypath = []
+        for level in derivations:
+            if level == "m":
+                continue
+            elif level[-1] == '\'':
+                keypath.append(int(level[:-1]) + HARDENED)
+            else:
+                keypath.append(int(level))
+        if len(keypath) > 1:
+            if keypath[1] == 1 + HARDENED:
+                is_mainnet = False
+
+        return keypath, is_mainnet
+
+    def get_xpub(self, bip32_path, xtype):
+        print(xtype)
+        xpub_keypath, is_mainnet = self.bip32_list_from_keypath_string(bip32_path)
+        if xtype == "p2wpkh":
+            if is_mainnet:
+                coin_network = bitbox02.btc.BTC
+                out_type = bitbox02.btc.BTCPubRequest.ZPUB
+            else:
+                coin_network = bitbox02.btc.TBTC
+                out_type = bitbox02.btc.BTCPubRequest.VPUB
+        elif xtype == "p2wpkh-p2sh":
+            if is_mainnet:
+                coin_network = bitbox02.btc.BTC
+                out_type = bitbox02.btc.BTCPubRequest.YPUB
+            else:
+                coin_network = bitbox02.btc.TBTC
+                out_type = bitbox02.btc.BTCPubRequest.UPUB
+        elif xtype == "standard":
```

Comment:
what does standard refer to? p2pkh? in this case we don't support it and should also be an exception?

Reply:
This is needed as a workaround to get the fingerprint. Should be removed again once that functionality is implemented.

Reply:
It is still needed now for the wallet encryption.

Reply:
Fixed with yet another hack :P